specs/time_series_api.adoc

```diff
@@ -0,0 +1,285 @@
+        addition of its apecic interface as a plugin and doesn't require
+        touching any other part of the code.
+    *** The way the plugins work is :
+      **** The interfaces to time-series db are added under the directory
+           tendrl/performance-monitoring/time_series_db/dbplugins of
+           tendrl/performance-monitoring.
+      **** These plugins inherit from a common interface and hence all of them
+           follow a similar basic skeletal structure.
+      **** The plugin framework uses python's meta-class approach and they get
+           registered with the framework when everything in the plugin path is
+           imported but only the plugin configured in trendrl's conf will be
+           used to interface to the time-series db. The way this works is as
+           follows:
+            ***** A singleton manager looks for all/any plugins in directory
+                  tendrl/performance-monitoring/time_series_db/dbplugins and
+                  imports them all.
+            ***** The moment they are imported the meta-class registers an
+                  instance of the plugin with itself.
+            ***** The singleton manager then stores the instance of required
+                  time-series db plugin which will be henceforth used for all
+                  interactions with the time-series db.
+            ***** Since the manager is a singleton, there is only one instance
+                  of it throughout the life of Tendrl/performance-monitoring
+                  application which means all this house-keeping is done only
+                  once during the lifecycle of the application.
+* The following apis will be exposed by Tendrl/performance-monitoring:
+  ** An api to list the monitored resources of a particular entity
+  ** An api to get statistics of any monitored resource from list in above api
+     for a particular entity.
+* The Tendrl/tendrl-api proxies to these apis and remains agnostic of the
+  time-series db specifics.
+
+=== Alternatives
+
+None
+
+=== Data model impact:
+
+Data in time-series db is stored as key-value pairs(utilization to timestamp)
+in its specific collection/table named by the fully qualified resource name
+and this is very specific to the way collectd's plugins send metrics to time
+series db and hence not much control can be exercised here unless we use
+custom collectd plugins rather than the collectd provided ones
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+The tendrl-api interfaces to the apis mentioned above under the section
+"Proposed change" and detailed below under "Notifications/Monitoring impact".
+
+==== Notifications/Monitoring impact:
+
+The api list to be exposed will grow with time. Currently, the following apis
+can be considered:
+
+To explore the resources that are monitored, we can use the api described
+below.
+
+Sample Request
+
+----------
+GET /monitoring/nodes/<node_name>/monitored_types
```

Comment:
Basically, the problem is that, the stats are stored in graphite with node names and it is the collectd plugin that decides the name to be associated with a metric its like chosen prefix.node_name.resource_type.stat_type . If only we override the custom collectd plugins and write our own plugins, we'll be able to customize this name. But anyway instead of the api user having to do the mapping, the performance_monitoring application can as well do it using details in etcd. I'll modify the spec to reflect this